src/components/SearchForm/SearchForm.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import styles from "./SearchForm.module.scss";
+
+const SearchForm: React.FC<{
+  InitialSearchQuery: string;
+  onSearch: (searchQuery: string) => void;
+}> = ({ InitialSearchQuery, onSearch }) => {
+  const [searchQuery, setSearchQuery] = useState(InitialSearchQuery);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+    console.log("handleInputChange", event.target.value);
+  };
+
+  const handleSearchClick = () => {
+    if (onSearch) {
+      onSearch(searchQuery);
+      console.log("handleSearchClick", searchQuery);
+    }
+  };
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      event.preventDefault();
+      handleSearchClick();
+    }
+  };
+
+  return (
+    <form>
+      <div className={styles["search"]}>
+        <input
+          type="text"
+          value={searchQuery}
+          onChange={handleInputChange}
+          onKeyDown={handleKeyPress}
+          placeholder="What do you want to watch?"
+          className={styles["search__input"]}
+        />
+        <button
+          type="button"
```

Comment:
This is a form, and this button in fact submits it, so it should work as a submit button. If it's a single button in the form, it will have "submit" type by default, so you don't need to provide it at all.